File: csharp/ToolGood.Algorithm/Internals/Functions/Flow/Function_ISNUMBER.cs
using System;
using System.Text;

namespace ToolGood.Algorithm.Internals.Functions.Flow
{
	internal class Function_ISNUMBER : Function_1
    {
        public Function_ISNUMBER(FunctionBase[] func1) : base(func1)
        {
        }

        public override string Name => "IsNumber";

        public override Operand Evaluate(AlgorithmEngine work, Func<AlgorithmEngine, string, Operand> tempParameter)
        {
            var args1 = func1.Evaluate(work, tempParameter);
            if (args1.IsNumber) { return Operand.True; }
            return Operand.False;
        }

    }

}

File: java/src/main/java/toolgood/algorithm/internals/functions/flow/Function_ISNUMBER.java
package toolgood.algorithm.internals.functions.flow;

import toolgood.algorithm.Internals.Operand;
import toolgood.algorithm.Internals.AlgorithmEngine;
import toolgood.algorithm.internals.FunctionBase;

public class Function_IsNumber extends Function_1 {
    public Function_IsNumber(FunctionBase func1) {
        super(func1);
    }

    @Override
    public Operand Evaluate(AlgorithmEngine work, java.util.function.Function<AlgorithmEngine, String, Operand> tempParameter) {
        Operand args1 = func1.Evaluate(work, tempParameter);
        if (args1.IsNumber()) {
            return Operand.True();
        }
        return Operand.False();
    }

    @Override
    public void toString(java.lang.StringBuilder stringBuilder, boolean addBrackets) {
        AddFunction(stringBuilder, "IsNumber");
    }
}
